src/manifolds/Rotations.jl

```diff
@@ -240,37 +240,40 @@ end
 
 flat!(M::Rotations, ξ::CoTFVector, p, X::TFVector) = copyto!(ξ, X)
 
-function get_coordinates(M::Rotations, p, X, B::ArbitraryOrthonormalBasis) where {N}
-    T = Base.promote_eltype(p, X)
-    return vee(M, p, X) .* sqrt(T(2))
-end
-
-function get_vector(M::Rotations, p, X, B::ArbitraryOrthonormalBasis) where {N}
-    T = Base.promote_eltype(p, X)
-    return hat(M, p, X) ./ sqrt(T(2))
-end
-
 @doc raw"""
-    hat(M::Rotations, p, Xⁱ)
+    get_vector(M::Rotations, p, Xⁱ)
 
 Convert the unique tangent vector components `Xⁱ` at point `p` on [`Rotations`](@ref)
 group $\mathrm{SO}(n)$ to the matrix representation $X$ of the tangent vector. See
-[`vee`](@ref vee(::Rotations, ::Any...)) for the conventions used.
+[`get_coordinates`](@ref get_coordinates(::Rotations, ::Any...)) for the conventions used.
 """
-hat(::Rotations, ::Any...)
+get_vector(::Rotations, ::Any...)
+function get_vector(M::Rotations{2}, p, Xⁱ::Real, B::DefaultBasis)
+    X = zeros(2,2)
+    return get_vector!(M, X, p, Xⁱ, B)
+end
+get_vector(M::Rotations{2}, p, Xⁱ, B::DefaultBasis) = get_vector(M, p, Xⁱ[1], B)
+function get_vector(M::Rotations{N}, p, Xⁱ, B::DefaultBasis) where {N}
+    X = zeros(N,N)
+    return get_vector!(M, X, p, Xⁱ, B)
+end
+function get_vector(M::Rotations, p, X, B::DefaultOrthonormalBasis) where {N}
+    T = Base.promote_eltype(p, X)
+    return get_vector(M, p, X, DefaultBasis()) ./ sqrt(T(2))
+end
 
-function hat!(M::Rotations{2}, X, p, Xⁱ::Real)
+get_vector!(M::Rotations{2}, X, p, Xⁱ, B::DefaultBasis) = get_vector!(M, X, p, Xⁱ[1], B)
```

Comment:
I don't know what the solution is, but it feels a bit odd seeing the mutating form of `get_vector!` used here when it isn't used in `bases.jl`.
Why for `get_vector` and `get_coordinates` do we not follow our usual pattern of the mutating function being wrapped in the non-mutating?

Reply:
Yes, we could do that, indeed.

Reply:
I just did mutating versions of `get_coordinates` and `get_vector` and introduced our usual pattern. For the basis I have to think a little bit more about the corresponding allocate.

Reply:
Initially I wasn't particularly concerned about allocations here so I just didn't write the mutating variants. But just for consistency it is worth changing to the usual pattern.

Reply:
Oh the tests still fail, I did not run all tests while travelling; might only come back to that on Thursday then.